pkg/deploy/deploy.go

```diff
@@ -244,6 +253,11 @@ func removeChildren(ctx context.Context, parent, root graph.ConfigNode, configGr
 }
 
 func deployConfig(ctx context.Context, c *config.Config, clientset *client.ClientSet, resolvedEntities config.EntityLookup) (entities.ResolvedEntity, error) {
+	if concurrentDeploymentsLimiter != nil {
```

Comment:
The caller should be responsible for handling that.
Since you limit this on a clientSet level, I think it would be better to put it  higher in the code.